include/RevProc.h

```diff
@@ -730,45 +759,45 @@ class RevProc{
   /// RevProc: decode a compressed CJ-type isntruction
   RevInst DecodeCJInst(uint16_t Inst, unsigned Entry) const;
 
-  /// RevProc: determine if the instruction is floating-point
-  bool IsFloat(unsigned Entry) const {
-    // Note: This is crude and looks for ANY FP register operands;
-    // InstTable[...].r<reg>Class should be used when doing hazard
-    // detection on particular registers, since some instructions
-    // combine integer and FP register operands. See DependencySet().
-    return( InstTable[Entry].rdClass  == RevRegClass::RegFLOAT ||
-            InstTable[Entry].rs1Class == RevRegClass::RegFLOAT ||
-            InstTable[Entry].rs2Class == RevRegClass::RegFLOAT ||
-            InstTable[Entry].rs3Class == RevRegClass::RegFLOAT );
-  }
-
   /// RevProc: Determine next thread to execute
   unsigned GetNextHartToDecodeID() const;
 
   /// RevProc: Whether any scoreboard bits are set
-  bool AnyDependency(unsigned HartID, bool isFloat) const {
+  bool AnyDependency(unsigned HartID,
+                     RevRegClass regClass = RevRegClass::RegUNKNOWN) const {
     const RevRegFile* regFile = GetRegFile(HartID);
-    return isFloat ? regFile->FP_Scoreboard.any() : regFile->RV_Scoreboard.any();
-  }
-
-  /// RevProc: Whether any scoreboard bits are set
-  bool AnyDependency(unsigned HartID) const {
-    const auto& RegFile = Harts.at(HartID)->RegFile;
-    return RegFile->FP_Scoreboard.any() || RegFile->RV_Scoreboard.any();
+    switch(regClass){
+      case RevRegClass::RegGPR:
+        return regFile->RV_Scoreboard.any();
+      case RevRegClass::RegFLOAT:
+        return regFile->FP_Scoreboard.any();
+      case RevRegClass::RegUNKNOWN:
+        return regFile->RV_Scoreboard.any() || regFile->FP_Scoreboard.any();
+      default:
+        return false;
```

Comment:
This should be an error condition - an invalid RegClass was passed

Reply:
I thought that if an "invalid" or "unknown" `RevRegClass` was passed, the dependency checker should return `false` as the default. If a register class is unknown or invalid, it does not have any dependencies as far as this check goes.

There is a `RegIMM` class which I have thought about removing (replacing instances of it with `RegUNKNOWN`), but which I wanted to discuss with you first.